tests/jballerina-unit-test/src/test/resources/test-src/expressions/unaryoperations/unary-operation.bal

```diff
@@ -324,6 +324,39 @@ function testNullableUnaryExpressions() {
     assertEquality(a19, 10);
 }
 
+function testResultingTypeOfUnaryPlus() {
+    int:Unsigned8 a = 10;
+    a = +a;
```

Comment:
I would use a separate variable instead, in a way to differentiate between the operation being performed on the operand itself vs an operation that makes use of the value of the operand.

Reply:
+1